docs/interfacing_with_surrogates.rst

```diff
@@ -0,0 +1,250 @@
+      params[f"Dense_{i}"] = layer_dict
+
+    params = {'params': params}
+
+
+The model can then be called like any Flax model,
+
+.. code-block:: python
+
+    output_tensor = flax_model.apply(params, input_tensor)
+
+
+Option 2: converting a Pytorch model to a JAX model
+===================================================
+
+.. warning::
+    The `torch_xla2`_ package is still evolving, which means there may be unexpected breaking changes. Some of the methods described in this section may become deprecated with little warning.
+
+If your model is in PyTorch, you could also consider using the `torch_xla2`_ package to do the conversion to JAX automatically.
+
+.. code-block:: python
+
+    import torch
+    import torch_xla2 as tx
+
+    trained_model = torch.load(PYTORCH_MODEL_PATH, weights_only=False) # Use weights_only=False if you want to load the full model
+    params, jax_model_from_torch = tx.extract_jax(model)
+
+The model can then be called as a pure JAX function:
+
+.. code-block:: python
+
+    output_tensor = jax_model_from_torch(params, input_tensor)
+
+To remove the need for performing the conversion every time the model is loaded, you might want to save a JAX-compatible version of the weights and model to disk:
+
+.. code-block:: python
+
+    import jax
+    import numpy as np
+
+    # jax.export uses StableHLO to serialize the model to a binary format
+    exported_model = jax.export(jax_model_from_torch)
+    with open("model.hlo", "wb") as f:
+      f.write(exported_model.serialize())
+
+    # The weights can be saved as numpy arrays
+    np.savez("weights.npz", *params)
+
+The model can then be loaded and run as follows:
+
+.. code-block:: python
+
+    # Load the HLO checkpoint
+    with open('model.hlo', 'rb') as f:
+      model_as_bytes = f.read()
+      model = jax.export.deserialize(model_as_bytes)
+
+    # Load the weights
+    weights_as_npz = jnp.load('weights.npz')
+    weights = [jnp.array(v) for v in weights_as_npz.values()]
+
+
+Option 3: using an ONNX model
+=============================
+
+The `Open Neural Network Exchange`_ format (ONNX) is a highly interoperable format for sharing neural network models. ONNX files include the model architecture and weights bundled together.
+
+An ONNX model can be loaded and called as follows, making sure to specify the correct input and output node names for your specific model:
+
+.. code-block:: python
+
+    import onnxruntime as ort
+    import numpy as np
+
+    s = ort.InferenceSession(ONNX_MODEL_PATH)
+    onnx_output_tensor = s.run(
+      # Output node names
+      ['output1', 'output2'],
+      # Mapping from input node names to input tensors
+      # NOTE: input tensors must have correct dtype for your specific model
+      {'input': np.asarray(input_tensor, dtype=np.float32)},
+    )
+
+However, JAX will not be able to differentiate through the InferenceSession.
+To convert the ONNX model to a JAX representation, you can use the `jaxonnxruntime`_ package:
```

Comment:
Maybe use hyperlink to jaxonnxruntime

Reply:
This is a hyperlink, I believe. Checked by compiling the docs locally.